plugins/browser-plugin-media-tracking/src/index.ts

```diff
@@ -0,0 +1,230 @@
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+import {
+  isTypeTextTrackEvent,
+  isTypeDocumentEvent,
+  boundryErrorHandling,
+  trackingOptionsParser,
+} from './helperFunctions';
+import { SnowplowMediaEvent } from './snowplowEvents';
+import { DocumentEvent, MediaEvent } from './mediaEvents';
+import { MediaEventType, TrackingOptions, RecievedTrackingOptions, EventGroup, trackedElement } from './types';
+import { BrowserPlugin, BrowserTracker, dispatchToTrackersInCollection } from '@snowplow/browser-tracker-core';
+import { buildSelfDescribingEvent, CommonEventProperties, SelfDescribingJson } from '@snowplow/tracker-core';
+import { MediaPlayerEvent } from './contexts';
+import { findMediaElem } from './findMediaElement';
+import { buildMediaEvent } from './buildMediaEvent';
+import { MediaProperty } from './mediaProperties';
+
+declare global {
+  interface HTMLVideoElement {
+    autoPictureInPicture: boolean;
+    disableRemotePlayback: boolean;
+    disablePictureInPicture: boolean;
+  }
+  interface HTMLAudioElement {
+    disableRemotePlayback: boolean;
+  }
+  interface Document {
+    pictureInPictureElement: Element;
+  }
+}
+
+const _trackers: Record<string, BrowserTracker> = {};
+
+export function MediaTrackingPlugin(): BrowserPlugin {
+  return {
+    activateBrowserPlugin: (tracker: BrowserTracker) => {
+      _trackers[tracker.id] = tracker;
+    },
+  };
+}
+
+const trackedIds: Record<string, trackedElement> = {};
+
+export function enableMediaTracking(args: { id: string; options?: RecievedTrackingOptions }) {
+  let conf: TrackingOptions = trackingOptionsParser(args.id, args.options);
+
+  const eventsWithOtherFunctions: Record<string, Function> = {
+    [DocumentEvent.FULLSCREENCHANGE]: (el: HTMLAudioElement | HTMLVideoElement, conf: TrackingOptions) => {
+      if (document.fullscreenElement?.id === args.id) {
+        mediaPlayerEvent(el, DocumentEvent.FULLSCREENCHANGE, conf);
+      }
+    },
+    [MediaEvent.SEEKED]: (el: HTMLAudioElement | HTMLVideoElement, conf: TrackingOptions) => {
+      if (conf.captureEvents.indexOf(SnowplowMediaEvent.PERCENTPROGRESS) !== 0) {
+        while (conf.progress!.boundryTimeoutIds.length) {
+          clearTimeout(Number(conf.progress!.boundryTimeoutIds.pop()));
+        }
+        setPercentageBoundTimeouts(el, conf);
+      }
+    },
+  };
+
+  const eventHandlers: Record<string, Function> = {};
+  for (let ev of conf.captureEvents) {
+    if (eventsWithOtherFunctions.hasOwnProperty(ev)) {
+      eventHandlers[ev] = (el: HTMLAudioElement | HTMLVideoElement) => eventsWithOtherFunctions[ev](el);
+    }
+    eventHandlers[ev] = (el: HTMLAudioElement | HTMLVideoElement, e: MediaEventType) => mediaPlayerEvent(el, e, conf);
+  }
+
+  trackedIds[args.id] = { searchIntervals: [], searchLimit: 5, tracking: false };
```

Comment:
is `searchLimit` the number of retries? `retryLimit` might be more descriptive or `searchRetryLimit`